src/utils/viewer/simviewer.cpp

```diff
@@ -0,0 +1,668 @@
+    timeSinceLastSimulation = 0.0;
+  }
+
+  // TODO: enable other sensors to be displayed
+  simulator_->displayObservation(0, "rgba_camera");
+
+  // TODO: not hooked up to Simulator
+  /*
+  if (debugBullet_) {
+    Mn::Matrix4 camM(renderCamera_->cameraMatrix());
+    Mn::Matrix4 projM(renderCamera_->projectionMatrix());
+
+    physicsManager_->debugDraw(projM * camM);
+  }
+  */
+
+  // draw picked object
+  if (objectPickingHelper_->isObjectPicked()) {
+    // setup blending function
+    Mn::GL::Renderer::enable(Mn::GL::Renderer::Feature::Blending);
+
+    // rendering
+    esp::gfx::RenderCamera::Flags flags;
+    if (simulator_->isFrustumCullingEnabled()) {
+      flags |= esp::gfx::RenderCamera::Flag::FrustumCulling;
+    }
+    renderCamera_->draw(objectPickingHelper_->getDrawables(), flags);
+
+    Mn::GL::Renderer::disable(Mn::GL::Renderer::Feature::Blending);
+  }
+
+  imgui_.newFrame();
+
+  if (showFPS_) {
+    ImGui::SetNextWindowPos(ImVec2(10, 10));
+    ImGui::Begin("main", NULL,
+                 ImGuiWindowFlags_NoDecoration | ImGuiWindowFlags_NoBackground |
+                     ImGuiWindowFlags_AlwaysAutoResize);
+    ImGui::SetWindowFontScale(2.0);
+    ImGui::Text("%.1f FPS", Mn::Double(ImGui::GetIO().Framerate));
+    uint32_t total = simulator_->getActiveSceneGraph().getDrawables().size();
+    ImGui::Text("%u drawables", total);
+    // TODO: how to get number culled from sensor?
+    // ImGui::Text("%u culled", total - visibles);
+    ImGui::End();
+  }
+
+  /* Set appropriate states. If you only draw ImGui, it is sufficient to
+     just enable blending and scissor test in the constructor. */
+  Mn::GL::Renderer::enable(Mn::GL::Renderer::Feature::Blending);
+  Mn::GL::Renderer::enable(Mn::GL::Renderer::Feature::ScissorTest);
+  Mn::GL::Renderer::disable(Mn::GL::Renderer::Feature::FaceCulling);
+  Mn::GL::Renderer::disable(Mn::GL::Renderer::Feature::DepthTest);
+
+  imgui_.drawFrame();
+
+  /* Reset state. Only needed if you want to draw something else with
+     different state after. */
+
+  Mn::GL::Renderer::enable(Mn::GL::Renderer::Feature::DepthTest);
+  Mn::GL::Renderer::enable(Mn::GL::Renderer::Feature::FaceCulling);
+  Mn::GL::Renderer::disable(Mn::GL::Renderer::Feature::ScissorTest);
+  Mn::GL::Renderer::disable(Mn::GL::Renderer::Feature::Blending);
+
+  swapBuffers();
+  timeline_.nextFrame();
+  redraw();
+}
+
+void SimViewer::viewportEvent(ViewportEvent& event) {
+  auto& sensors = simulator_->getAgent(0)->getSensorSuite();
+  for (auto entry : sensors.getSensors()) {
+    auto visualSensor =
+        dynamic_cast<esp::sensor::VisualSensor*>(entry.second.get());
+    if (visualSensor != nullptr) {
+      visualSensor->specification()->resolution = {event.windowSize()[1],
+                                                   event.windowSize()[0]};
+      simulator_->getRenderer()->bindRenderTarget(*visualSensor);
+      Cr::Utility::Debug() << visualSensor->framebufferSize();
+    }
+  }
```

Comment:
Image is still stretched on canvas resize. Thoughts?

Reply:
Can we not preserve aspect ratio somehow?

Reply:
Take a look at #322 and #331. 
I believe you will have to set the aspectRatioPolicy::Extend for the renderCamera.

Reply:
Thanks, this worked @bigbike!